apps/src/eval/eval.js

```diff
@@ -228,6 +233,60 @@ function getDrawableFromBlockXml(blockXml) {
 }
 
 /**
+ * Parse an EvalObject looking for EvalText objects. For each one, extract the
+ * text content.
+ */
+Eval.getTextStringsFromObject_ = function (evalObject) {
+  if (!evalObject) {
+    return [];
+  }
+
+  var strs = [];
+  if (evalObject instanceof EvalText) {
+    strs.push(evalObject.getText());
+  }
+
+  evalObject.getChildren().forEach(function (child) {
+    strs = strs.concat(Eval.getTextStringsFromObject_(child));
+  });
+  return strs;
+};
+
+/**
+ * @returns True if two eval objects have sets of text strings that differ
+ *   only in case
+ */
+Eval.haveCaseMismatch_ = function (object1, object2) {
+  var strs1 = Eval.getTextStringsFromObject_(object1);
+  var strs2 = Eval.getTextStringsFromObject_(object2);
+
+  if (strs1.length !== strs2.length) {
+    return false;
+  }
+
+  strs1.sort();
+  strs2.sort();
+
+  var mismatch = false;
+
+  for (var i = 0; i < strs1.length; i++) {
+    var str1 = strs1[i];
+    var str2 = strs2[i];
+    if (str1.toLowerCase() !== str2.toLowerCase()) {
```

Comment:
Yeah, it felt a little strange to me as I was writing it too. Your way looks better.